### PR TITLE
Replace artificial forests, part III

### PR DESCRIPTION
**Pull request by FarkasKBence**
#### Summary
Content "Updates more locations that didn't use regional terrain/predecessor mapgen"

#### Purpose of change

To create smoother transitions between forests and specials.  Simplifies code, merges maps, replaces forest clutter with predecessor_mapgen, and adds regional terrain to certain locations to reduce their file size and make them easier to read/edit.

#### Describe the solution

The following locations were edited:

* **boat rental:** moved to lakeshores, palettized, defined regional terrain
* **botanic garden:** added regional terrain, added more plants
* **fishing pond:** merged maps, palettized, defined predecessor mapgen
* **nature trail:** merged maps, palettized, defined predecessor mapgen
* **private park:** added regional terrain, defined predecessor mapgen
* **public art piece:** added regional terrain
* **public pond:** merged maps, palettized, defined predecessor mapgen
* **public space:** added regional terrain
* **shooting range:** merged maps, added regional terrain
* **small wooded trail:** palettized, defined predecessor mapgen
* **state park:** merged maps, palettized, defined predecessor mapgen
* **sugar house:** added regional terrain, replaced wooden floor with concrete in the kitchen
* **tree farm:** merged maps, added regional terrain

#### Describe alternatives you've considered

None.

#### Testing

Surveyed the land for the aforementioned locations. Works as expected.

#### Additional context

<details>
  <summary>Screenshots:</summary>

**boat rental**
rental

**botanical garden**
botgard

**fishing pond**
fishingpond

**nature trail**
naturetrail

**small wooded trail**
trailsmall

**state park**
statepark

</details>

An unwanted change is that small wooded trails and nature trails have way fewer trees now, even if I change the predecessor mapgen from `forest` to `forest_thick`.  Probably related to #54040?
Also, I haven't updated my game in a while, but placing locations on lakeshore via predecessor mapgen still produces buggy results.  Boat rental only look correct when they're facing north.

### CODE REVIEW

File: data/json/mapgen/cs_private_park.json [hunk 1]
```diff
@@ -5,95 +5,76 @@
...
+        "B": "t_floor",
+        "E": "t_floor",
+        "L": "t_floor",
+        "S": "t_floor",
+        "s": "t_sidewalk",
+        "|": "t_wall_wood",
         "D": "t_door_locked_peep",
         "C": "t_door_c",
         "G": "t_curtains",
-        "P": "t_tree_cherry",
```
Comment (NetSysFire) [inline data/json/mapgen/cs_private_park.json:47]: Honestly it look like choice of trees might have been intentional here. A private park harboring fancy trees (which may include fruit trees) makes sense but I have no idea how you could specify that without hardcoding locations.

File: data/json/mapgen/cs_private_park.json [hunk 2]
```diff
@@ -5,95 +5,76 @@
     "om_terrain": [ "cs_private_park" ],
     "weight": 1000,
     "object": {
-      "fill_ter": "t_floor",
+      "predecessor_mapgen": "forest_thick",
```
Comment (NetSysFire) [inline data/json/mapgen/cs_private_park.json:8]: Dense forest may not be appropriate for a park since it is way too thick. Maybe just use `forest`?

**Review by NetSysFire — APPROVED** [review]
Nice work! Thank you so much for working on this and continuing to make all those structures much cleaner and better to work with, especially for new contributors which may be overwhelmed by the ascii soup being a hardcoded forest. These are all much needed changes.

File: data/json/mapgen/cs_private_park.json (hunk 1 above)
Comment (LovamKicsiGazsii) [reply data/json/mapgen/cs_private_park.json:47]: privatepark
Re-added willow and cherry trees.

**Pull request merged**